src/lib/core/data-table/data-table.ts

```diff
@@ -196,6 +215,32 @@ export class CdkTable<T> implements CollectionViewer {
     }
 
     this._isViewInitialized = true;
+
+    this._StickyRows.changes.subscribe((queryList: QueryList<CdkStickyRow>) => {
```

Comment:
This wasn't so much to re-bootstrap anything if changes are made, but more so because this is the only way I could find to signal when the table is rendered enough to start doing height calculations :O.

Conceivably though if a user adds the "cdkStickyRow" directive to a row after the table is rendered this will re-adjust the table to accommodate that row (regardless of height or color).

If I am understanding the question correctly I do not believe the row's content should be allowed to override the table's set height (in case the table needs to be siloed to a particular area of the page). 

I believe the code adapts for rows of varying heights :).